cupy/random/generator.py

```diff
@@ -235,6 +235,74 @@ def uniform(self, low=0.0, high=1.0, size=None, dtype=float):
         rand = self.random_sample(size=size, dtype=dtype)
         return dtype.type(low) + rand * dtype.type(high - low)
 
+    def choice(self, a, size=None, replace=True, p=None):
+        """Returns an array of random values from a given 1-D array.
+
+        .. seealso::
+            :func:`cupy.random.choice` for full document,
+            :meth:`numpy.random.choice`
+
+        """
+        if a is None:
+            raise ValueError('a must be 1-dimensional or an integer')
+        if isinstance(a, six.integer_types):
+            a_size = a
+            if a_size <= 0:
+                raise ValueError('a must be greater than 0')
+            a = cupy.array(a, copy=False)
+        else:
+            a = cupy.array(a, copy=False)
+            if a.ndim != 1:
+                raise ValueError('a must be 1-dimensional')
+            elif a.dtype != 'int':
+                raise NotImplementedError
+            else:
+                a_size = len(a)
+                if a_size == 0:
+                    raise ValueError('a must be non-empty')
+
+        if p is not None:
+            p = cupy.array(p)
+            if p.ndim != 1:
+                raise ValueError('p must be 1-dimensional')
+            if len(p) != a_size:
+                raise ValueError('a and p must have same size')
+            if not (p >= 0).all():
+                raise ValueError('probabilities are not non-negative')
+            p_sum = cupy.sum(p).get()
+            if not numpy.allclose(p_sum, 1):
+                raise ValueError('probabilities do not sum to 1')
+
+        if not replace:
+            raise NotImplementedError
+
+        if size is None:
+            raise NotImplementedError
+        shape = size
+        size = numpy.prod(shape)
+
+        if p is not None:
+            p = cupy.broadcast_to(p, (size, a_size))
+            index = cupy.argmax(cupy.log(p) -
+                                cupy.random.gumbel(size=(size, a_size)),
+                                axis=1)
+            index = index.astype(cupy.int32, copy=False)
```

Comment:
Could you add a comment why we need this line (e.g. to align the dtype with NumPy).